VideoWeb/VideoWeb/ClientApp/src/app/security/authentication/quick-link-security.service.ts

```diff
@@ -0,0 +1,145 @@
+import { JwtHelperService } from '../jwt-helper.service';
+import { ISecurityService } from './security-service.interface';
+
+export class QuickLinkJwtBody extends JWTBody {
+    // tslint:disable-next-line: variable-name
+    preferred_username: string;
+    private exp: number;
+
+    constructor(body: any) {
+        super(body);
+    }
+
+    get expiry(): number {
+        return this.exp;
+    }
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class QuickLinkSecurityService implements ISecurityService {
+    private loggerPrefix = '[QuickLinkSecurityService] -';
+    private token: string;
+    private tokenSessionStorageKey = 'QUICK_LINKS_JWT';
+    private tokenSessionStorage: SessionStorage<string>;
+    private isAuthenticatedSubject = new BehaviorSubject<boolean>(false);
+    private userDataSubject = new ReplaySubject<any>(1);
+
+    decodedTokenBody: QuickLinkJwtBody;
+
+    constructor(private apiClient: ApiClient, private jwtHelper: JwtHelperService) {
+        this.tokenSessionStorage = new SessionStorage<string>(this.tokenSessionStorageKey);
+        this.token = this.tokenSessionStorage.get();
+        if (this.isTokenValid(this.token)) {
+            this.isAuthenticatedSubject.next(true);
+        }
+    }
+
+    authorize(authOptions?: AuthOptions, token?: string): void {
+        this.setToken(token);
+        this.checkAuth().pipe(take(1)).subscribe();
+    }
+
+    private clearToken() {
+        this.setToken(null);
+    }
+
+    private setToken(token: string | null) {
+        if (token === null) {
+            this.token = null;
+            this.decodedTokenBody = null;
+            this.tokenSessionStorage.clear();
+            this.isAuthenticatedSubject.next(false);
+        } else {
+            this.token = token;
+            this.decodedTokenBody = this.decodeTokenBody(this.token);
+            this.tokenSessionStorage.set(token);
+        }
+    }
+
+    checkAuth(): Observable<boolean> {
+        const tokenIsAuthorisedResult = source =>
+            new Observable<boolean>(subscriber => {
+                return source.subscribe({
+                    next() {
+                        subscriber.next(true);
+                    },
+                    error() {
+                        subscriber.next(false);
+                    },
+                    complete() {
+                        subscriber.complete();
+                    }
+                });
+            });
+        return this.apiClient.isQuickLinkParticipantAuthorised().pipe(
+            tokenIsAuthorisedResult,
+            tap(authenticated => {
+                this.isAuthenticatedSubject.next(authenticated);
+
+                // TODO: Due to AppInsightsLoggerService provided in app.module injecting Logger causes a circular dependency. This should be refactored
+                if (authenticated) {
+                    console.debug(`${this.loggerPrefix} Check auth passed. User is authenticated.`);
```

Comment:
replace with logger?

Reply:
Afraid this caused a circular dependency when I tried this before. I would like to figure out a way to prevent this, but that would probably take some time to resolve.